packages/discord.js/src/structures/ChannelSelectMenuInteraction.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+const { Collection } = require('@discordjs/collection');
+const MessageComponentInteraction = require('./MessageComponentInteraction');
+
+/**
+ * Represents a {@link ComponentType.ChannelSelect} select menu interaction.
+ * @extends {MessageComponentInteraction}
+ */
+class ChannelSelectMenuInteraction extends MessageComponentInteraction {
+  constructor(client, data) {
+    super(client, data);
+
+    /**
+     * Collection of the selected channels
+     * @type {Collection<Snowflake, Channel|APIChannel>}
+     */
+    this.channels = new Collection();
+    for (const channel of Object.values(data.data.resolved.channels)) {
+      this.channels.set(channel.id, this.client.channels._add(channel, this.guild) ?? channel);
```

Comment:
should we cache the channels received? I think not, right @discordjs/core?

Reply:
Yes. I did it this way because so do slash commands - space agreed it's sound logic to make it consistent.